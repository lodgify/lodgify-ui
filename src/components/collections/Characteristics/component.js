import React from 'react';
import PropTypes from 'prop-types';
import { List } from 'semantic-ui-react';

import { buildKeyFromStrings } from 'utils/build-key-from-strings';
import { getFirstFourItems } from 'utils/get-first-four-items';
import { Grid } from 'layout/Grid';
import { GridColumn } from 'layout/GridColumn';
import { Icon } from 'elements/Icon';
import { ShowOn } from 'layout/ShowOn';

/**
 * The standard widget for displaying a collection of characteristics for a property or a room.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ spaceCharacteristics }) => (
  <Grid columns={1}>
    <GridColumn width={12}>
      <ShowOn
        computer
        parent={List}
        parentProps={{ horizontal: true }}
        tablet
        widescreen
      >
        {getFirstFourItems(spaceCharacteristics).map(
          ({ iconName, text }, index) => (
            <List.Item key={buildKeyFromStrings(text, index)}>
              <Icon labelText={text} name={iconName} />
            </List.Item>
          )
        )}
      </ShowOn>
      <ShowOn mobile parent={Grid} parentProps={{ columns: 1 }}>
        {getFirstFourItems(spaceCharacteristics).map(
          ({ iconName, text }, index) => (
            <GridColumn
              computer={3}
              key={buildKeyFromStrings(text, index)}
              mobile={6}
            >
              <Icon labelText={text} name={iconName} />
            </GridColumn>
          )
        )}
      </ShowOn>
    </GridColumn>
  </Grid>
);

Component.displayName = 'Characteristics';

Component.defaultProps = {
  spaceCharacteristics: [],
};

Component.propTypes = {
  /** The main characteristics to display. Maximum four. */
  spaceCharacteristics: PropTypes.arrayOf(
    PropTypes.shape({
      /**
       * The name of the icon to display.
       * [See here for the full list of valid icon names](https://github.com/lodgify/lodgify-ui/blob/master/src/components/elements/Icon/constants.js)
       */
      iconName: PropTypes.string.isRequired,
      /** A visible label to display for the key fact. */
      text: PropTypes.string.isRequired,
    })
  ),
};
