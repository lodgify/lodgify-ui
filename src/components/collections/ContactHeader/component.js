import React from 'react';
import { Menu } from 'semantic-ui-react';
import {
  string,
  arrayOf,
  shape,
  oneOfType,
  oneOf,
  number,
  bool,
  func,
} from 'prop-types';

import { FlexContainer } from 'layout/FlexContainer';
import { Button } from 'elements/Button';
import { Submenu } from 'elements/Submenu';
import { size } from 'utils/size';
import { Icon, ICON_NAMES } from 'elements/Icon';
import { ShowOn } from 'layout/ShowOn';
import { testidFactory } from 'utils/testid';
import { Logo } from 'elements/Logo';

import { HorizontalMenu } from '../HorizontalMenu';

import { MobileHeader } from './MobileHeader';
import { PHONE_PREFIX } from './constants';

const TEST_ID_PREFIX = 'contactHeader';

const testid = testidFactory(TEST_ID_PREFIX);

/**
 * A header displays a logo, grouped navigation items
 * contact information and an a primary call to action.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  primaryButtonText,
  navigationMenuItems,
  languageOptions,
  onChangeLanguage,
  currencyOptions,
  onChangeCurrency,
  currencyValue,
  currencyNoResultsText,
  languageValue,
  phoneNumber,
  logoHref,
  logoSubText,
  logoSize,
  logoSrc,
  logoSizes,
  logoText,
  logoSrcSet,
}) => (
  <header className="contact-header" data-testid={testid()}>
    <ShowOn computer tablet widescreen>
      <FlexContainer alignItems="center" justifyContent="space-between">
        <Menu>
          {size(languageOptions) > 1 && (
            <Menu.Item>
              <Submenu
                items={languageOptions}
                name="language"
                onChange={onChangeLanguage}
                value={languageValue}
                willOpenAbove
              />
            </Menu.Item>
          )}
          {size(currencyOptions) > 1 && (
            <Menu.Item>
              <Submenu
                isSearchable
                items={currencyOptions}
                name="currency"
                noResultsText={currencyNoResultsText}
                onChange={onChangeCurrency}
                value={currencyValue}
                willOpenAbove
              />
            </Menu.Item>
          )}
          {!!phoneNumber && (
            <Menu.Item className="is-selectable">
              <a href={`${PHONE_PREFIX}${phoneNumber.replace(/\s/g, '')}`}>
                <Icon labelText={phoneNumber} name={ICON_NAMES.PHONE} />
              </a>
            </Menu.Item>
          )}
        </Menu>
        <Logo
          logoHref={logoHref}
          logoSize={logoSize}
          logoSizes={logoSizes}
          logoSrc={logoSrc}
          logoSrcSet={logoSrcSet}
          logoSubText={logoSubText}
          logoSubTextAlignment="center"
          logoText={logoText}
        />
        <Button>{primaryButtonText}</Button>
      </FlexContainer>
      <HorizontalMenu isHeader items={navigationMenuItems} />
    </ShowOn>
    <ShowOn mobile>
      <FlexContainer
        alignItems="center"
        className="hidden-contact-header"
        justifyContent="space-between"
      >
        <Logo
          logoHref={logoHref}
          logoSize={logoSize}
          logoSizes={logoSizes}
          logoSrc={logoSrc}
          logoSrcSet={logoSrcSet}
          logoSubText={logoSubText}
          logoText={logoText}
        />
        <MobileHeader
          currencyNoResultsText={currencyNoResultsText}
          currencyOptions={currencyOptions}
          currencyValue={currencyValue}
          languageOptions={languageOptions}
          languageValue={languageValue}
          logoHref={logoHref}
          logoSize={logoSize}
          logoSizes={logoSizes}
          logoSrc={logoSrc}
          logoSrcSet={logoSrcSet}
          logoSubText={logoSubText}
          logoText={logoText}
          navigationItems={navigationMenuItems}
          onChangeCurrency={onChangeCurrency}
          phoneNumber={phoneNumber}
        />
      </FlexContainer>
    </ShowOn>
  </header>
);

Component.displayName = 'ContactHeader';

Component.defaultProps = {
  currencyNoResultsText: undefined,
  currencyOptions: [],
  currencyValue: null,
  languageOptions: [],
  languageValue: null,
  logoHref: '',
  logoSize: 'medium',
  logoSizes: undefined,
  logoSrc: undefined,
  logoSrcSet: undefined,
  logoSubText: '',
  logoText: '',
  onChangeCurrency: undefined,
  onChangeLanguage: undefined,
  phoneNumber: '',
  primaryButtonText: '',
};

Component.propTypes = {
  /** The text to display when no results are returned from the currency dropdown. */
  currencyNoResultsText: string,
  /** The options which the user can select for the currency dropdown. */
  currencyOptions: arrayOf(
    shape({
      /** The visible text for the option. */
      text: string.isRequired,
      /** The underlying value for the option. */
      value: oneOfType([bool, number, string]),
    })
  ),
  /** The current value of the currency dropdown. */
  currencyValue: oneOfType([bool, number, string]),
  /** The options which the user can select for the language dropdown. */
  languageOptions: arrayOf(
    shape({
      /** The visible text for the option. */
      text: string.isRequired,
      /** The underlying value for the option. */
      value: oneOfType([bool, number, string]),
    })
  ),
  /** The current value of the language dropdown. */
  languageValue: oneOfType([bool, number, string]),
  /** The href for the logo link. */
  logoHref: string,
  /** The maximum size of the headers logo. */
  logoSize: oneOf(['medium', 'large', 'huge']),
  /** A list of one or more strings separated by commas indicating a set of source sizes for the logo. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
  logoSizes: string,
  /** The src url for the logo. */
  logoSrc: string,
  /** A list of one or more strings separated by commas indicating a set of possible image sources for the user agent to use for the logo. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
  logoSrcSet: string,
  /** The text that appears under the logo or logo text. */
  logoSubText: string,
  /** The text for the logo. */
  logoText: string,
  /** The items for a user to navigate the site. */
  navigationMenuItems: arrayOf(
    shape({
      /** The href url for an item which is a link. */
      href: string,
      /** Sub navigation items to display as a [`<Submenu />`](#submenu) under an item. */
      subItems: arrayOf(
        shape({
          href: string.isRequired,
          target: string,
          text: string.isRequired,
        })
      ),
      /** Specifies where to display the linked navigation items URL. See [MDN docs `<a />` for more](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-target). */
      target: string,
      /** The visible text for an item. */
      text: string.isRequired,
    })
  ).isRequired,
  /** The function called when the currency dropdown is changed.
   *  @param {String} name
   *  @param {String} value
   */
  onChangeCurrency: func,
  /** The function called when the language dropdown is changed.
   *  @param {String} name
   *  @param {String} value
   */
  onChangeLanguage: func,
  /** The phone number to display. */
  phoneNumber: string,
  /** The text to display for the primary button. */
  primaryButtonText: string,
};
