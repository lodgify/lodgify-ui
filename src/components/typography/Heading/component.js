import React from 'react';
import PropTypes from 'prop-types';
import { Header } from 'semantic-ui-react';

import { getHeadingNumber } from './getHeadingNumber';

const SIZES = ['huge', 'large', 'medium', 'small'];

/**
 * A heading identifies a block of content.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  children,
  className,
  isColorInverted,
  size,
  textAlign,
}) => (
  <Header
    as={`h${getHeadingNumber(SIZES, size)}`}
    className={className}
    inverted={isColorInverted}
    textAlign={textAlign}
  >
    {children}
  </Header>
);

Component.displayName = 'Heading';

Component.defaultProps = {
  className: null,
  isColorInverted: false,
  textAlign: null,
  size: 'medium',
};

Component.propTypes = {
  /** The heading text. */
  children: PropTypes.string.isRequired,
  /**
   * Custom class name.
   * Provided by `ShowOn` so ignored in the styleguide.
   * @ignore
   */
  className: PropTypes.string,
  /** Is the color of the heading inverted for contrast. */
  isColorInverted: PropTypes.bool,
  /** The size of the heading. */
  size: PropTypes.oneOf(SIZES),
  /** The alignment of the heading text. */
  textAlign: PropTypes.oneOf(['left', 'right', 'justified', 'center']),
};
