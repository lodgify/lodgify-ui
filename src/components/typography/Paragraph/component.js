import React from 'react';
import PropTypes from 'prop-types';
import getClassNames from 'classnames';

const MEDIUM = 'medium';
const TINY = 'tiny';

/**
 * A paragraph provides text content
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ children, isCompact, size, weight }) => (
  <p
    className={getClassNames(weight, {
      tiny: size === TINY,
      'is-compact': isCompact,
    })}
  >
    {children}
  </p>
);

Component.displayName = 'Paragraph';

Component.defaultProps = {
  isCompact: false,
  size: MEDIUM,
  weight: null,
};

Component.propTypes = {
  /** The paragraph content. */
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  /** The paragraph has reduced line height. */
  isCompact: PropTypes.bool,
  /** The size of the paragraph. */
  size: PropTypes.oneOf([MEDIUM, TINY]),
  /** The weight of the paragraph. */
  weight: PropTypes.oneOf(['heavy', 'light']),
};
