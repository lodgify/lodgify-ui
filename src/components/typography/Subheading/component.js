import React from 'react';
import PropTypes from 'prop-types';

/**
 * A subheading provides concise information about a block of content.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ children }) => (
  <span className="ui sub header">{children}</span>
);

Component.displayName = 'Subheading';

Component.propTypes = {
  /** The subheading text. */
  children: PropTypes.string.isRequired,
};
