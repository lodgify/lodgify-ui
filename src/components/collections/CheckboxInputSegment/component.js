import React from 'react';
import PropTypes from 'prop-types';
import { Segment } from 'semantic-ui-react';

import { Divider } from 'elements/Divider';
import { FlexContainer } from 'layout/FlexContainer';
import { Grid } from 'layout/Grid';
import { Heading } from 'typography/Heading';
import { Checkbox } from 'inputs/Checkbox';
import { GridColumn } from 'layout/GridColumn';
import { withResponsive } from 'utils/with-responsive';
import { buildKeyFromStrings } from 'utils/build-key-from-strings';

import { getGridColumns } from './utils/getGridColumns';

/**
 * A collection of information alongside checkboxes.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
const Component = ({ heading, checkboxes, isUserOnMobile }) => (
  <Segment className="is-input-segment is-checkbox-input-segment" vertical>
    <FlexContainer
      alignItems="baseline"
      flexDirection="column"
      justifyContent="space-between"
    >
      <Heading>{heading}</Heading>
      <Divider size="small" />
      <Grid columns={getGridColumns(isUserOnMobile)}>
        {checkboxes.map(({ name, label, ...checkboxProps }) => (
          <GridColumn key={buildKeyFromStrings(label, name)}>
            <Checkbox
              label={label}
              name={name}
              {...checkboxProps}
              isFluid={isUserOnMobile}
              isLabelLeft={isUserOnMobile}
            />
          </GridColumn>
        ))}
      </Grid>
    </FlexContainer>
  </Segment>
);

Component.displayName = 'CheckboxInputSegment';

Component.propTypes = {
  /** The checkboxes to display. See [the `Checkbox` component for valid props](#/Inputs/Checkbox).  */
  checkboxes: PropTypes.arrayOf(PropTypes.object).isRequired,
  /** The text for the input segment's heading. */
  heading: PropTypes.string.isRequired,
  /**
   * Is the user on a mobile device.
   * Provided by `withResponsive` so ignored in the styleguide.
   * @ignore
   */
  isUserOnMobile: PropTypes.bool.isRequired,
};

export const ComponentWithResponsive = withResponsive(Component);
