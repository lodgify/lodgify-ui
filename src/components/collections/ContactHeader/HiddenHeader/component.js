import React, { useRef, useState, useEffect } from 'react';
import {
  string,
  arrayOf,
  shape,
  func,
  bool,
  oneOfType,
  number,
} from 'prop-types';
import { Menu, Accordion } from 'semantic-ui-react';
import { size } from 'lodash';

import { Heading } from 'typography/Heading';
import { buildKeyFromStrings } from 'utils/build-key-from-strings';
import { Icon, ICON_NAMES } from 'elements/Icon';
import { Modal } from 'elements/Modal';
import { Submenu } from 'elements/Submenu';
import { testidFactory } from 'utils/testid';

import { PHONE_PREFIX } from '../constants';

const TEST_ID_PREFIX = 'hiddenHeader';

const testid = testidFactory(TEST_ID_PREFIX);

export const Component = ({
  headingHref,
  headingText,
  navigationItems,
  languageOptions,
  onChangeLanguage,
  languageValue,
  currencyOptions,
  currencyNoResultsText,
  onChangeCurrency,
  currencyValue,
  phoneNumber,
}) => {
  const [activeItem, setActiveItem] = useState('');

  useEffect(() => {
    /* istanbul ignore next */
    if (navigationItems.length > 0) {
      setActiveItem(navigationItems[0].text);
    }
  }, [navigationItems.length]);

  const menu = useRef();

  const onMenuLinkClick = text => {
    setActiveItem(text);
  };

  return (
    <Modal
      className="hidden-contact-header-modal"
      closeIcon={<Icon labelText="Close" name="close" />}
      data-testid={testid()}
      header={
        <Menu.Item href={headingHref} link>
          <Heading className="header-title" size="small">
            {headingText}
          </Heading>
        </Menu.Item>
      }
      isFullscreen
      trigger={<Icon name={ICON_NAMES.BARS} />}
    >
      <div ref={menu}>
        <Menu text vertical>
          {navigationItems.map(({ subItems, text, link }, index) =>
            size(subItems) ? (
              <Accordion
                as={Menu.Item}
                key={buildKeyFromStrings(text, index)}
                panels={[
                  {
                    title: {
                      content: text,
                      key: buildKeyFromStrings(text, index),
                    },
                    content: {
                      content: subItems.map(({ text, href }, index) => (
                        <Menu.Item
                          active={text === activeItem}
                          data-testid={testid(`accordionItem_${index}`)}
                          href={href}
                          key={buildKeyFromStrings(text, index)}
                          link
                          onClick={() => onMenuLinkClick(text)}
                        >
                          {text}
                        </Menu.Item>
                      )),
                      key: buildKeyFromStrings(index, text),
                    },
                  },
                ]}
              />
            ) : (
              <Menu.Item
                active={text === activeItem}
                data-testid={testid(`hiddenHeaderItem_${index}`)}
                href={link}
                key={buildKeyFromStrings(text, index)}
                link
                onClick={() => onMenuLinkClick(text)}
              >
                {text}
              </Menu.Item>
            )
          )}
        </Menu>
        <Menu vertical>
          {size(languageOptions) > 1 && (
            <Menu.Item>
              <Submenu
                items={languageOptions}
                name="language"
                onChange={onChangeLanguage}
                value={languageValue}
                willOpenAbove
              />
            </Menu.Item>
          )}
          {size(currencyOptions) > 1 && (
            <Menu.Item>
              <Submenu
                isSearchable
                items={currencyOptions}
                name="currency"
                noResultsText={currencyNoResultsText}
                onChange={onChangeCurrency}
                value={currencyValue}
                willOpenAbove
              />
            </Menu.Item>
          )}
          {!!phoneNumber && (
            <Menu.Item className="is-selectable">
              <a href={`${PHONE_PREFIX}${phoneNumber.replace(/\s/g, '')}`}>
                <Icon labelText={phoneNumber} name={ICON_NAMES.PHONE} />
              </a>
            </Menu.Item>
          )}
        </Menu>
      </div>
    </Modal>
  );
};

Component.defaultProps = {
  currencyNoResultsText: '',
  currencyOptions: [],
  currencyValue: null,
  headingHref: '',
  headingText: '',
  languageOptions: [],
  languageValue: null,
  phoneNumber: '',
  onChangeCurrency: undefined,
  onChangeLanguage: undefined,
};

Component.propTypes = {
  /** The text to display when no results are returned from the currency dropdown. */
  currencyNoResultsText: string,
  /** The options which the user can select for the currency dropdown. */
  currencyOptions: arrayOf(
    shape({
      /** The visible text for the option. */
      text: string.isRequired,
      /** The underlying value for the option. */
      value: oneOfType([bool, number, string]),
    })
  ),
  /** The current value of the currency dropdown. */
  currencyValue: oneOfType([bool, number, string]),
  /** The link for the heading. */
  headingHref: string,
  /** The text for the heading. */
  headingText: string,
  /** The options which the user can select for the language dropdown. */
  languageOptions: arrayOf(
    shape({
      /** The visible text for the option. */
      text: string.isRequired,
      /** The underlying value for the option. */
      value: oneOfType([bool, number, string]),
    })
  ),
  /** The current value of the language dropdown. */
  languageValue: oneOfType([bool, number, string]),
  /** The items for a user to navigate the site. */
  navigationItems: arrayOf(
    shape({
      /** The href url for an item which is a link. */
      href: string,
      /** Sub navigation items to display as a [`<Submenu />`](#submenu) under an item. */
      subItems: arrayOf(
        shape({
          href: string.isRequired,
          target: string,
          text: string.isRequired,
        })
      ),
      /** Specifies where to display the linked navigation items URL. See [MDN docs `<a />` for more](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-target). */
      target: string,
      /** The visible text for an item. */
      text: string.isRequired,
    })
  ).isRequired,
  /** The function called when the currency dropdown is changed.
   *  @param {String} name
   *  @param {String} value
   */
  onChangeCurrency: func,
  /** The function called when the language dropdown is changed.
   *  @param {String} name
   *  @param {String} value
   */
  onChangeLanguage: func,
  /** The phone number to display. */
  phoneNumber: string,
};
