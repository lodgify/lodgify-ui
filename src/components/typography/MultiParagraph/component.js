import React, { Fragment, useState } from 'react';
import { string, bool } from 'prop-types';
import classnames from 'classnames';

import { SHOW_MORE } from 'utils/default-strings';
import { testidFactory } from 'utils/testid';
import { getParagraphsFromStrings } from 'utils/get-paragraphs-from-strings';
import { Paragraph } from 'typography/Paragraph';

const TEST_ID_PREFIX = 'multiparagraph';

const testid = testidFactory(TEST_ID_PREFIX);

export const Component = ({ content, showMoreLabel, isShowingAll }) => {
  const [isOpen, setIsOpen] = useState(isShowingAll);
  const paragraphs = getParagraphsFromStrings(content);

  return (
    <Fragment>
      <div
        className={classnames('multi-paragraph', { open: isOpen })}
        data-testid={testid()}
      >
        {paragraphs.map((paragraph, key) => (
          <Paragraph data-testid={testid(`paragraph-${key}`)} key={key}>
            {paragraph}
          </Paragraph>
        ))}
      </div>
      {!isOpen && (
        <div
          className="multi-paragraph-read-more"
          data-testid={testid('button-read-more')}
          onClick={() => setIsOpen(true)}
          role="button"
        >
          <span>{showMoreLabel}</span>
        </div>
      )}
    </Fragment>
  );
};

Component.displayName = 'MultiParagraph';
Component.defaultProps = {
  showMoreLabel: SHOW_MORE,
  isShowingAll: false,
};
Component.propTypes = {
  /** text to show or hide */
  content: string.isRequired,
  /** not cut the text as default */
  isShowingAll: bool,
  /** show more button label */
  showMoreLabel: string,
};
