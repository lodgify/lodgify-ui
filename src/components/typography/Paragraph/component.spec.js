import React from 'react';
import { mount } from 'enzyme';
import { expectComponentToHaveDisplayName } from '@lodgify/enzyme-jest-expect-helpers';

import { Component as Paragraph } from './component';

const children = ['ðŸš¸', 2];

const getParagraph = props =>
  mount(<Paragraph children={children} {...props} />);

describe('<Paragraph />', () => {
  describe('by default', () => {
    it('should render the right structure', () => {
      const actual = getParagraph();

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if `props.isCompact` is passed', () => {
    it('should render the right structure', () => {
      const actual = getParagraph({ isCompact: true });

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if `props.size` === tiny', () => {
    it('should render the right structure', () => {
      const actual = getParagraph({ size: 'tiny' });

      expect(actual).toMatchSnapshot();
    });
  });

  it('should have displayName `Paragraph`', () => {
    expectComponentToHaveDisplayName(Paragraph, 'Paragraph');
  });
});
