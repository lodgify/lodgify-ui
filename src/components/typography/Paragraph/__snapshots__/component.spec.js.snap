// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> by default should render the right structure 1`] = `
<Paragraph
  isCompact={false}
  size="medium"
  weight={null}
>
  <p
    className=""
  >
    ðŸš¸
    2
  </p>
</Paragraph>
`;

exports[`<Paragraph /> if \`props.isCompact\` is passed should render the right structure 1`] = `
<Paragraph
  isCompact={true}
  size="medium"
  weight={null}
>
  <p
    className="is-compact"
  >
    ðŸš¸
    2
  </p>
</Paragraph>
`;

exports[`<Paragraph /> if \`props.size\` === tiny should render the right structure 1`] = `
<Paragraph
  isCompact={false}
  size="tiny"
  weight={null}
>
  <p
    className="tiny"
  >
    ðŸš¸
    2
  </p>
</Paragraph>
`;
